contracts/core/EntryPoint.sol

```diff
@@ -230,16 +230,18 @@ contract EntryPoint is IEntryPoint, StakeManager {
         (address aggregator, uint256 deadline) = _validatePrepayment(0, userOp, outOpInfo, SIMULATE_FIND_AGGREGATOR);
         uint256 prefund = outOpInfo.prefund;
         uint256 preOpGas = preGas - gasleft() + userOp.preVerificationGas;
-        DepositInfo memory depositInfo = getDepositInfo(outOpInfo.mUserOp.paymaster);
-        PaymasterInfo memory paymasterInfo = PaymasterInfo(depositInfo.stake, depositInfo.unstakeDelaySec);
+        StakeInfo memory paymasterInfo = getStakeInfo(outOpInfo.mUserOp.paymaster);
+        StakeInfo memory senderInfo = getStakeInfo(outOpInfo.mUserOp.sender);
+        bytes calldata initCode = userOp.initCode;
+        address factory = initCode.length>=20 ? address(bytes20(initCode[0 : 20])) : address(0);
+        StakeInfo memory factoryInfo = getStakeInfo(factory);
```

Comment:
Both `factory` and `aggregator` can be `address(0)` and yet the code to handle them is very different. Also, only aggregator code is wrapped in another struct, making it nested. Is there a reason why this is done this way? I though the idea was to "unify" the code.

Reply:
Actually, the `paymaster` can also be `address(0)`, can it not?

Reply:
the reason for the difference is that paymaster and factory are specified in the userOp, but aggregator is a method call on the wallet.
so when simulate returns, it can tell what paymaster the stake relates to, but can't easily tell what aggregator was used (with deployed wallet it can make another eth_call to determine the aggregator, but with newly deployed wallet, it can't do even that) - so we must return it.